File: database/src/test/java/com/hartwig/actin/database/dao/DataUtilTest.java
package com.hartwig.actin.database.dao;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.List;
import java.util.stream.Stream;

import org.junit.Test;

public class DataUtilTest {

    @Test
    public void shouldConvertToByte() {
        assertThat(DataUtil.toByte(true)).isEqualTo((byte) 1);
        assertThat(DataUtil.toByte(false)).isEqualTo((byte) 0);
        assertThat(DataUtil.toByte(null)).isNull();
    }

    @Test
    public void shouldConcatStrings() {
        assertThat(DataUtil.concat(List.of("hi"))).isEqualTo("hi");
        assertThat(DataUtil.concat(List.of("hi", "hello"))).isEqualTo("hi;hello");
        assertThat(DataUtil.concat(null)).isNull();
    }

    @Test
    public void shouldConvertStreamToString() {
        assertThat(DataUtil.concatStream(Stream.of("first", "second"))).isEqualTo("first;second");
    }

    @Test
    public void shouldConvertNullableToString() {
        assertThat(DataUtil.nullableToString(null)).isNull();
        assertThat(DataUtil.nullableToString("test")).isEqualTo("test");
    }
}

File: database/src/test/kotlin/com/hartwig/actin/database/dao/DataUtilTest.kt
package com.hartwig.actin.database.dao

import com.hartwig.actin.database.dao.DataUtil.concat
import com.hartwig.actin.database.dao.DataUtil.concatObjects
import com.hartwig.actin.database.dao.DataUtil.nullableToString
import com.hartwig.actin.database.dao.DataUtil.toByte
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DataUtilTest {

    @Test
    fun `Should convert boolean to byte`() {
        assertThat(toByte(true)).isEqualTo(1.toByte())
        assertThat(toByte(false)).isEqualTo(0.toByte())
        assertThat(toByte(null)).isNull()
    }

    @Test
    fun `Should concat and sort strings`() {
        assertThat(concat(listOf("hi"))).isEqualTo("hi")
        assertThat(concat(listOf("hi", "hello"))).isEqualTo("hello;hi")
        assertThat(concat(null)).isNull()
    }

    @Test
    fun `Should concat and sort objects`() {
        assertThat(concatObjects(listOf(TestObject("hi")))).isEqualTo("TestObject(value=hi)")
        assertThat(concatObjects(listOf(TestObject("hi"), TestObject("hello")))).isEqualTo("TestObject(value=hello);TestObject(value=hi)")
    }

    @Test
    fun `Should convert nullable to string`() {
        assertThat(nullableToString(null)).isNull()
        assertThat(nullableToString("test")).isEqualTo("test")
    }

    data class TestObject(val value: String)
}